c++/kernel_overhead_cl/Mandelbrot.cpp

```diff
@@ -0,0 +1,187 @@
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+#include <vector>
+#include <cassert>
+#include <boost/format.hpp>
+
+#ifdef _WIN32
+#include <sys/timeb.h>
+#include <windows.h>
+#else
+#include <sys/time.h>
+#include <unistd.h>
+#endif
+
+#include "TGA.h"
+#include "oclutils.h"
+
+inline double getCurrentTime() {
+#ifdef WIN32
+    LARGE_INTEGER f;
+    LARGE_INTEGER t;
+    QueryPerformanceFrequency(&f);
+    QueryPerformanceCounter(&t);
+    return t.QuadPart / (double)f.QuadPart;
+#else
+    struct timeval tv;
+    struct timezone tz;
+    gettimeofday(&tv, &tz);
+    return tv.tv_sec + tv.tv_usec*1e-6;
+#endif
+};
+
+std::vector<std::vector<float> > mandelbrot(
+    unsigned int nx, unsigned int ny, unsigned int iterations,
+    std::vector<float> x0, std::vector<float> y0,
+    std::vector<float> dx, std::vector<float> dy,
+    size_t block_width = 8, size_t block_height = 8) {
+    cl_int error = CL_SUCCESS;
+    int num_zooms = x0.size();
+    assert(num_zooms == x0.size());
+    assert(num_zooms == y0.size());
+    assert(num_zooms == dx.size());
+    assert(num_zooms == dy.size());
+
+    //Allocate GPU data 
+    std::vector<cl::Buffer> output_gpu(num_zooms);
+    for (int i = 0; i < num_zooms; ++i) {
+        output_gpu[i] = cl::Buffer(*OpenCLUtils::getContext(), CL_MEM_READ_WRITE,
+            nx*ny*sizeof(float), NULL, &error);
+        CL_CHECK(error);
+    }
+
+    //Create timing events
+    std::vector<cl::Event> events(num_zooms);
```

Comment:
Please remove these. The kernel should have profilingi n formation so we do not need to enqueue these events also (relevant because we are timing the kernels both on CPU and PGU side)